code/modules/mob/living/silicon/say.dm

```diff
@@ -10,24 +10,41 @@
 	if(issilicon(src))
 		var/mob/living/silicon/S = src
 		desig = trim_left(S.designation + " " + S.job)
-	var/message_a = say_quote(message)
-	var/rendered = "Robotic Talk, <span class='name'>[name]</span> <span class='message'>[message_a]</span>"
+	var/large_message_a = say_quote(message, list("robot big"))
+	var/message_a = say_quote(message, list("robot"))
 	for(var/mob/M in GLOB.player_list)
 		if(M.binarycheck())
 			if(isAI(M))
-				var/renderedAI = "<span class='srt_radio binarysay'>Robotic Talk, <a href='?src=[REF(M)];track=[html_encode(name)]'><span class='name'>[name] ([desig])</span></a> <span class='message'>[message_a]</span></span>"
-				to_chat(M, renderedAI)
+				var/source = FALSE
+				if(M == src)
+					source = TRUE
+				var/rendered = "<span class='srt_radio binarysay'>Robotic Talk, <a href='?src=[REF(M)];track=[html_encode(name)]'><span class='name'>[name] ([isAI(M) ? "[desig]" : ""])</span></a> <span>[source ? "[large_message_a]" : "[message_a]"]</span>"
+				to_chat(M, rendered) //AI hears only itself on loud mode.
+			else if(iscyborg(M))
+				var/mob/living/silicon/robot/borg = M
+				var/source = FALSE
+				if(src == borg.connected_ai)
+					source = TRUE
+				var/rendered = "<span class='srt_radio binarysay'>Robotic Talk, <a href='?src=[REF(M)];track=[html_encode(name)]'><span class='name'>[name]</span></a> <span>[source ? "[large_message_a]" : "[message_a]"]</span>"
```

Comment:
You're giving cyborg AI tracking.
The original message was
`"Robotic Talk, <span class='name'>[name]</span> <span class='message'>[message_a]</span>"`

Note: I don't see any point of using `<span class='message'>`, so it seems to be fine to remove.